charts/govuk-apps-conf/templates/router-configmap.yaml

```diff
@@ -32,6 +32,36 @@ data:
         return lc($r->uri);
       }';
 
+      # Set GOVUK-Request-Id if not set
+      # See http://nginx.org/en/docs/http/ngx_http_perl_module.html
+      perl_modules perl/lib;
+      perl_set $govuk_request_id '
+        sub {
+          my $r = shift;
+          my $current_header = $r->header_in("GOVUK-Request-Id");
+          if (defined $current_header && $current_header ne "") {
+            return $current_header;
+          } else {
+            my $pid = $r->variable("pid");
+            my $msec = $r->variable("msec");
+            my $remote_addr = $r->variable("remote_addr");
+            my $request_length = $r->variable("request_length");
+            return "$pid-$msec-$remote_addr-$request_length";
+          }
+        }
+      ';
+
+      # This map creates a $sts_default variable for later use.
+      # If this header is already set by upstream, then $sts_default will
+      # be an empty string, which will later lead to:
+      #    add_header Strict-Transport-Security ''
+      # which will be ignored according to http://serverfault.com/a/598106
+      # If the header is not set by upstream, then $sts_default will be set
+      # and later uses in add_header will be effective.
+      map $upstream_http_strict_transport_security $sts_default {
```

Comment:
So I understand this just comes verbatim from the old config, so let's assume for now that it works as intended 😅 I agree it's a good idea to keep it as-is for now and leave potential improvements for maybe later 👍